src/main/java/io/aiven/kafka/auth/AivenAclAuthorizer.java

```diff
@@ -138,108 +138,69 @@ public boolean authorize(final Session session,
-        if (principalType.equals(KafkaPrincipal.USER_TYPE)) {
-            cacheKey = resource + "|" + operation + "|" + principalName + "|" + principalType;
-        }
 
-        // we loop here until we can evaluate the access with fresh configuration
-        while (true) {
-            // First, check if we have a fresh config, and if so, evaluate access request
-            lock.readLock().lock();
-            try {
-                if (lastUpdateCheckTimestamp + 10000 > now) {
-                    if (cacheKey != null && verdictCache != null) {
-                        final Boolean cachedVerdict = verdictCache.get(cacheKey);
-                        if (cachedVerdict != null) {
-                            verdict = cachedVerdict.booleanValue();
-                            logAuthVerdict(verdict, operation, resource, principalType, principalName, true);
-                            return verdict;
-                        }
-                    }
-
-                    final Iterator<AivenAcl> iter = aclEntries.iterator();
-                    while (!verdict && iter.hasNext()) {
-                        final AivenAcl aclEntry = iter.next();
-                        if (aclEntry.check(principalType, principalName, operation, resource)) {
-                            verdict = true;
-                        }
-                    }
-                    logAuthVerdict(verdict, operation, resource, principalType, principalName, false);
-                    if (cacheKey != null && verdictCache != null) {
-                        verdictCache.put(cacheKey, verdict);
-                    }
-                    return verdict;
-                }
-            } finally {
-                lock.readLock().unlock();
-            }
-
-            // We may need to update the config
-            lock.writeLock().lock();
-            try {
-                // Recheck the timer, as an another thread may have updated config
-                // while we waited for the lock.
-                if (lastUpdateCheckTimestamp + 10000 <= now) {
-                    lastUpdateCheckTimestamp = now;
-                    checkAndUpdateConfig();
-                }
-            } finally {
-                lock.writeLock().unlock();
-            }
-        }
+        final boolean verdict = cacheReference.get().get(principalType, principalName, operation, resource);
+        logAuthVerdict(verdict, operation, resource, principalType, principalName);
+        return verdict;
     }
 
     private void logAuthVerdict(final boolean verdict,
                                 final String operation,
                                 final String resource,
                                 final String principalType,
-                                final String principalName,
-                                final boolean cached) {
-        final String cachedStr = cached ? " (cached)" : "";
+                                final String principalName) {
```

Comment:
I see you removed `cached`. I understand the reason (too much trouble passing it from the depth of `VerdictCache`) and we probably don't need this. However, let's be explicit about such things in the PR message and commits.